hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/AnalyzeStatementTest.java

```diff
@@ -192,6 +194,20 @@ public void test_changeClusterStateToPassive() {
                 .isInstanceOf(CancellationException.class);
     }
 
+    @Test
+    public void test_updateConfigForAnalyzedQuery() {
+        // When
+        Job job = runQuery();
+
+        assertThatThrownBy(() -> job.updateConfig(new DeltaJobConfig()))
+                .hasMessageContaining("is not suspendable, can't perform `updateJobConfig()`");
+
+        // Ensure job is running after the refusal to alter the job
+        assertJobStatusEventually(job, RUNNING);
```

Comment:
this check might be race-condition prone: if job termination in `updateConfig` were slow (if `updateConfig` terminated the job in similar way to `suspend`) you might not observe the new state yet. In other words `assertJobStatusEventually(job, RUNNING);` is not equivalent to "job was not terminated"

Reply:
Tests in `JobTest` (e.g, `test_tryUpdatingJobConfig_then_fail`) also use a similar check. I don't think there is a better way.

Reply:
partial solution is to reassert the condition after some time, eg. if the job was still running after 1s then (probably) it was not terminated. Of course, this slows the test down and is still not fully bullet-proof.